frontend/src/components/PortfolioFundingByCAN/PortfolioFundingByCAN.jsx

```diff
@@ -0,0 +1,133 @@
+import { useSelector } from "react-redux";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faSquare } from "@fortawesome/free-solid-svg-icons";
+import { useEffect, useState } from "react";
+
+import { ResponsiveDonutWithInnerPercent } from "../UI/ResponsiveDonutWithInnerPercent/ResponsiveDonutWithInnerPercent";
+
+import constants from "../../constants";
+import CustomLayerComponent from "../UI/ResponsiveDonutWithInnerPercent/CustomLayerComponent";
+import CurrencyWithSmallCents from "../UI/CurrencyWithSmallCents/CurrencyWithSmallCents";
+
+import cssClasses from "./styles.module.css";
+import RoundedBox from "../RoundedBox/RoundedBox";
+
+library.add(faSquare);
+
+const styles = {
+    root: {
+        width: 175,
+        height: 175,
+    },
+    cardBody: {
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "left",
+        borderColor: "#FFF",
+    },
+    cardGroup: {
+        display: "flex",
+        marginBottom: "7%",
+        flexDirection: "row",
+        justifyContent: "space-evenly",
+        width: "250px",
+    },
+    cardItem: {
+        display: "flex",
+        flex: "1",
+    },
+    iconStyle: {
+        verticalAlign: "middle",
+        paddingRight: "4px",
+    },
+};
+
+const PortfolioFundingByCAN = (props) => {
+    const portfolioFunding = useSelector((state) => state.portfolioFundingSummary.portfolioFunding);
+    const portfolioCansFundingDetails = useSelector((state) => state.portfolioDetail.portfolioCansFundingDetails);
+    const [percent, setPercent] = useState("");
+    const [canChartData, setCanChartData] = useState([]);
+
+    const cardBody = `padding-top-2 padding-left-4 ${styles.cardBody}`;
+
+    useEffect(() => {
+        const portfolio_total_funding = parseFloat(portfolioFunding.total_funding.amount);
+
+        if (portfolio_total_funding) {
+            const data = portfolioCansFundingDetails
+                .filter((item) => item.total_funding)
+                //eslint-disable-next-line array-callback-return
+                .map((item, index) => {
+                    if (item.total_funding) {
+                        return {
+                            id: item.can.id,
+                            value: item.total_funding,
+                            percent: `${Math.floor((parseFloat(item.total_funding) / portfolio_total_funding) * 100)}%`,
+                            color: constants.colors[index],
+                            number: item.can.number,
+                        };
+                    }
+                });
+
+            console.log(`data=${JSON.stringify(data)}`);
```

Comment:
is this intended to leave in perpetually ?

Reply:
removed